src/main/java/executor/service/config/beans/ThreadPoolFactory.java

```diff
@@ -0,0 +1,38 @@
+package executor.service.config.beans;
+
+import executor.service.model.ThreadPoolConfig;
+
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+
+/*
+Simple beans that creates, initializes and returns ThreadPoolExecutor.
+Later will be changed by adding @Bean annotation.
+ */
+public class ThreadPoolFactory {
+    private static final int MAXIMUM_POOL_SIZE = 16;
+    private static ThreadPoolExecutor executor;
+
+    public static ThreadPoolExecutor getThreadPoolExecutor() {
+        if (executor == null) {
+            synchronized (ThreadPoolConfig.class) {
```

Comment:
1. DI will create a singleton of this class and inject it. Why such an implementation? Where does the concurrency come from?
2. I would draw your attention to the fact that this class will be tested using PowerMockito, which is not very simple.